src/java.base/share/classes/java/lang/invoke/ValueBootstrapMethods.java

```diff
@@ -458,6 +458,35 @@ private static LinkageError newLinkageError(Throwable e) {
         return (LinkageError) new LinkageError().initCause(e);
     }
 
+    /**
+     * Invoke the bootstrap methods hashCode for the given instance.
+     * @param o the instance to hash.
+     * @return the hash code of the given instance {code o}.
+     */
+    public static int invokeHashCode(Object o) {
+        try {
+            Class<?> type = o.getClass();
+            // Note: javac disallows user to call super.hashCode if user implementated
+            // risk for recursion for experts crafting byte-code
+            return (int) hashCodeInvoker(type).invoke(o);
+        } catch (Error|RuntimeException e) {
+            throw e;
+        } catch (Throwable e) {
+            if (VERBOSE) e.printStackTrace();
+            throw new InternalError(e);
+        }
+    }
+
+    static <T> MethodHandle hashCodeInvoker(Class<T> type) {
+        return HASHCODE_METHOD_HANDLES.get(type);
+    }
+
+    private static ClassValue<MethodHandle> HASHCODE_METHOD_HANDLES = new ClassValue<>() {
+        @Override protected MethodHandle computeValue(Class<?> type) {
+            return MethodHandleBuilder.hashCodeForType(type);
```

Comment:
@MrSimms thanks for working on this.

I expect this to call the existing `hashCodeInvoker` if the type is an inline type.   `hashCodeForType` is the method handle invoking `java.util.Objects::hashCode`.   The existing `hashCodeInvoker` should be changed to take `Class<?> type` once javac no longer generates `equals` and `hashCode` method for inline type.

Maybe I should take this patch and make the coordinated changes??